src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/Worker.java

```diff
@@ -1272,6 +1297,10 @@ public Worker build() {
                 workerStateChangeListener = DEFAULT_WORKER_STATE_CHANGE_LISTENER;
             }
 
+            if(shardSyncer == null) {
+                shardSyncer = new ShardSyncer(new KinesisLeaseCleanupValidator());
```

Comment:
Don't take `ShardSyncer` as a option, take the `LeaseCleanupValidator` as an option.

Reply:
That would create multiple shardSyncer instances, I'll stick to the current state.